src/Core/src/Handlers/HybridWebView/HybridWebView.ts

```diff
@@ -0,0 +1,216 @@
+            sendMessageToDotNet('__RawMessage', message);
+        },
+
+        /*
+         * Invoke a .NET method on the InvokeJavaScriptTarget instance.
+         * The method name and parameters are serialized and sent to the .NET host application.
+         * 
+         * @param methodName The name of the .NET method to invoke.
+         * @param paramValues The parameters to pass to the .NET method. If the method takes no parameters, this can be omitted.
+         * 
+         * @returns A promise that resolves with the result of the .NET method invocation.
+         */
+        InvokeDotNet: async function (methodName: string, paramValues?: any) {
+            const body: JSInvokeMethodData = {
+                MethodName: methodName
+            };
+
+            // if parameters were provided, serialize them first
+            if (paramValues !== undefined) {
+                if (!Array.isArray(paramValues)) {
+                    paramValues = [paramValues];
+                }
+
+                for (let i = 0; i < paramValues.length; i++) {
+                    paramValues[i] = JSON.stringify(paramValues[i]);
+                }
+
+                if (paramValues.length > 0) {
+                    body.ParamValues = paramValues;
+                }
+            }
+
+            const message = JSON.stringify(body);
+
+            const requestUrl = `${window.location.origin}/__hwvInvokeDotNet?data=${encodeURIComponent(message)}`;
+
+            const rawResponse = await fetch(requestUrl, {
+                method: 'GET',
+                headers: {
+                    'Accept': 'application/json'
+                }
+            });
+            const response: DotNetInvokeResult = await rawResponse.json();
+
+            if (!response) {
+                return null;
+            }
+
+            if (response.IsJson) {
+                return JSON.parse(response.Result);
+            }
+
+            return response.Result;
+        },
+
+        /*
+         * Invoke a JavaScript method from the .NET host application.
+         * This method is called from the HybridWebViewHandler and is not intended to be used by user applications.
+         * 
+         * @param taskId The task ID that was provided by the .NET host application.
+         * @param methodName The JavaScript method to invoke in the global scope.
+         * @param args The arguments to pass to the JavaScript method.
+         * 
+         * @returns A promise.
+         */
+        __InvokeJavaScript: async function (taskId: string, methodName: Function, args: any[]) {
+            try {
+                let result: any = null;
+                // @ts-ignore - We are checking the type of the function here
+                if (methodName[Symbol.toStringTag] === 'AsyncFunction') {
+                    result = await methodName(...args);
+                } else {
+                    result = methodName(...args);
+                }
```

Comment:
You shouldn't need to do this check. In JavaScript, you can `await` any value. If you `await` a non-promise, then the `await` just no-ops and the expression is equivalent to its counterpart without the await.

```suggestion
                const result = await methodName(...args);
```